hs_core/views/utils.py

```diff
@@ -46,7 +46,8 @@ def upload_from_irods(username, password, host, port, zone, irods_fnames, res_fi
         fname = os.path.basename(ifname.rstrip(os.sep))
         res_files.append(UploadedFile(file=tmpFile, name=fname, size=size))
 
-def authorize(request, res_id, edit=False, view=False, full=False, superuser=False, raises_exception=True):
+def authorize(request, res_id, discoverable=False, edit=False, view=False,
```

Comment:
I am wondering whether we need to add an extra discoverable parameter to handle discoverable case specifically. Perhaps discoverable can be handled in similar way as public which is implicitly encoded in uaccess_can_view_resource()?